### DISCUSSION TITLE
Manager NAT yet again

### DISCUSSION THREAD

**Discussion by rosswakelin** [Security-Onion-Solutions/securityonion#10645] [Unsupported Versions]
I would like to lean upon the communities collective wisdom and experience.
We are about to roll out 30 sensor nodes across our distributed network, and they have to somehow get back to the manager that is sitting on the other side of a firewall in our core - the manager/data nodes etc. are sitting in a protected zone in our core that is not routable from the rest of the network.  We were going to NAT the manager address for the remote sensor nodes, but it seems from the other conversations here that this will not work because of the use of certificates with IP addresses.

Has anyone created this sort of distributed environment before, and how did you get around the NAT/IP/certificate issue.

Thanks

Ross

**Comment by petiepooo** [comment]
I would suggest using wireguard to create an overlay network where all sensors can connect to the manager/receiver nodes (and vice versa) via their wireguard interface.  It can be complicated to setup, but is very flexible and robust once it's working well.  If your certificate uses a FQDN, you can use the /etc/hosts file on the sensors to specify the manager's wireguard IP whereas DNS can still resolve to the normal administrative IP on the manager for analyst access.

At that size, I assume you're breaking out separate search and receiver functions (see #10647).  If your manager/receiver/search nodes are all in one subnet, it may make sense to have a dedicated wireguard server on that subnet just for routing.  You could then use static routes on manager/receiver nodes to send all sensor traffic to the wireguard server for forwarding on to sensor nodes.
